de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/util/ConceptActivationHelper.java

```diff
@@ -0,0 +1,219 @@
+		}
+		
+		// Now create the dependencies using the concept names as identifiers
+		for (Concept concept : selectedConcepts.values()) {
+			List<String> conceptDependencyIds = new ArrayList<>(ActiveConceptHelper.getConceptDependencies(concept));
+			dependencyTree.addDependencies(concept.getName(), conceptDependencyIds);
+		}
+		
+		List<String> orderedConcepts = dependencyTree.getLinearOrder();
+		
+		// And then install them by using the ordered list of concept names
+		// and the map of concept names pointing to the already preloaded concepts
+		for (String conceptName : orderedConcepts) {
+			if (selectedConcepts.containsKey(conceptName)) {
+				activateConcept(selectedConcepts.get(conceptName), editingDomain, progressMonitor);
+			}
+		}
+	}
+	
+	/**
+	 * Prepare a selected concept for activation. Checks if an older version is already in the repository and if so
+	 * migrates it to the latest version, otherwise the concept will be simply added to the active concepts
+	 * @param concept the selected concept
+	 * @param editingDomain the editing domain
+	 */
+	protected void activateConcept(Concept concept, EditingDomain editingDomain, IProgressMonitor progressMonitor) {
+
+		// Check if we already have this concept but with a different version added to the repository
+		Concept activeConcept = new ActiveConceptHelper(repository).getConcept(concept.getName());
+
+		boolean conceptIsInRepository = activeConcept != null;
+		
+		// There is a concept of a different version in the repository
+		if (conceptIsInRepository && !activeConcept.getVersion().equals(concept.getVersion())) {
+				
+			//Check if new dependencies have to be added before migration
+			if (activeConcept.eContainer() != null && activeConcept.eContainer() instanceof Repository) {
+				Repository repository = (Repository) activeConcept.eContainer();
+				new ConceptActivationHelper(repository).handleNewDependencies(activeConcept, editingDomain, progressMonitor);
+			}
+			
+			try {
+				Command migrateToLatestCommand = CreateMigrateConceptToLatestCommand.create(activeConcept, (TransactionalEditingDomain) editingDomain, progressMonitor);
+				editingDomain.getCommandStack().execute(migrateToLatestCommand);
+			} catch (CoreException e) {
+				DVLMEditPlugin.getPlugin().getLog().log(new Status(Status.ERROR, DVLMEditPlugin.PLUGIN_ID, "Failed to do migration on active concept: " + concept.getDisplayName(), e));
+			}
+			
+		}
+		
+		if (!conceptIsInRepository) {
+			Command cmd = ActiveConceptConfigurationElement.createCopyConceptToRepository(editingDomain, concept, repository);
+			editingDomain.getCommandStack().execute(cmd);
+		}
+	}
+	
+	/**
+	 * Activate new concept dependencies 
+	 * @param concept the concept to be prepared for migration
+	 * @param editingDomain the editing domain
+	 */
+	public void handleNewDependencies(Concept concept, EditingDomain editingDomain, IProgressMonitor progressMonitor) {

```

Comment:
Should we maybe also handle the situation when a dependency was removed in one of the concept versions?

Reply:
How would you handle that? I mean we can add that to the migrator, but we currently do not forsee removing deactivating concepts... So any idea what that would do with the removed dependency?

Reply:
Assume v1 has no dependency, v2 has a dependency, v3 has no dependency again. If I want to migrate from v1 to v3, I don't need to activate this dependency. But I guess it's not going to happen often, so we leave it